apps/apps.c

```diff
@@ -2633,6 +2634,7 @@ void wait_for_async(SSL *s)
         fds++;
     }
     select(width, (void *)&asyncfds, NULL, NULL, NULL);
+    OPENSSL_free(fds);
```

Comment:
since fds has been increased, this won't actually work. You'll need to keep the original pointer.